### PR TITLE
L.BingLayer: fixes / updates / refactoring

### PR DESCRIPTION
**Pull request by johnd0e**
Options:
- fix `culture` option for updated imagery sets
- fix `minZoom` / `maxZoom` options,
   add `applyMaxNativeZoom` method/option
- support `style` option
- support `detectRetina` option
- options are now compatible with https://github.com/digidem/leaflet-bing-layer

Constructor:
- `key` can be included in options

Misc updates and refactoring

---
This change is [Reviewable](https://reviewable.io/reviews/shramov/leaflet-plugins/288)

### CODE REVIEW

**Comment by johnd0e** [comment]
@brunob 

I've finished all planning changes here.
Please check and merge, if you are interested in L.BingLayer.
If not - just tell me, and I will setup dedicated repo.

**Comment by brunob** [comment]
Thx @johnd0e 'im totally intersetd in Bing layer, i'll check this asap :)

About dedicated repo, since many years i'm thinking about put every single script here in a dedicated repo, maybe it's the time to do it for Yandex first. We could do that in a common organization. Maybe @leaflet-extras could be the right place.

**Comment by johnd0e** [comment]
> maybe it's the time to do it for Yandex first.

Ok, I will notify you when I finish with Yandex.

> Maybe @leaflet-extras could be the right place.

I like this idea.

File: layer/tile/Bing.js [hunk 1]
```diff
@@ -1,131 +1,169 @@
...
 	options: {
-		subdomains: [0, 1, 2, 3],
-		type: 'Aerial',
+		// imagerySet: https://docs.microsoft.com/en-us/bingmaps/rest-services/imagery/get-imagery-metadata#template-parameters
+		// supported:
+		// - Aerial, AerialWithLabels (Deprecated), AerialWithLabelsOnDemand
+		// - Road (Deprecated), RoadOnDemand
+		// - CanvasDark, CanvasLight, CanvasGray
+		// not supported: Birdseye*, Streetside
+		type: 'RoadOnDemand',
```
Comment (brunob) [inline layer/tile/Bing.js]: default type should be `Aerial` in order to prevent changes on existings maps

Comment (johnd0e) [reply layer/tile/Bing.js]: I believe that 'RoadOnDemand' is better choice for default.
Besides, it's unlikely that even existing maps update themselves automatically, and every maintainer easily fix this option if needed.
So I propose consider this like major version change (and we just talked about repo change)

Comment (brunob) [reply layer/tile/Bing.js]: > I believe that 'RoadOnDemand' is better choice for default.

I believe that it's nicer for our users who simply do `L.BingLayer(apiKey)` that their map doesn't switch layer.

> and every maintainer easily fix this option if needed

and more easy to do nothing :)

File: layer/tile/Bing.js [hunk 2]
```diff
@@ -47,14 +60,19 @@ L.BingLayer = L.TileLayer.extend({
...
 			}
 			_this.initMetadata(meta);
 		};
-		var urlScheme = (document.location.protocol === 'file:') ? 'http' : document.location.protocol.slice(0, -1);
-		var url = urlScheme + '://dev.virtualearth.net/REST/v1/Imagery/Metadata/'
-					+ this.options.type + '?include=ImageryProviders&jsonp=' + cbid +
-					'&key=' + this._key + '&UriScheme=' + urlScheme;
+		var urlScheme = document.location.protocol === 'file:' ? 'http' :
+			document.location.protocol.slice(0, -1);
+		var url = urlScheme + '://dev.virtualearth.net/REST/v1/Imagery/Metadata/' + this.options.type;
```
Comment (brunob) [inline layer/tile/Bing.js]: we should ditch all this `urlScheme`things and simply use `htts` every time

File: layer/tile/Bing.js (hunk 1 above)
Comment (johnd0e) [reply layer/tile/Bing.js]: > and more easy to do nothing :)

Yeah, why update if all working just well in old good version?)
Ok, let it stay 'Aerial' here, but change when transfer to separate repo?

File: layer/tile/Bing.js (hunk 2 above)
Comment (johnd0e) [reply layer/tile/Bing.js]: I heard that some browsers forbid mixed content, if so - it may brake maps on http pages.

Comment (brunob) [reply layer/tile/Bing.js]: no, miwed content is only when http content is loaded from http webpage ;)

see https://github.com/leaflet-extras/leaflet-providers/pull/237

Comment (johnd0e) [reply layer/tile/Bing.js]: Ok, I will fix it.

**Comment by johnd0e** [comment]
'_Aerial*_ ' and deprecated imagery sets in fact do not have tiles for high zoom levels.
And there is no reliable way to get real maxZoom value via API (we always get `21`).
So I've just implemented heuristic one (will commit after testing).

**Comment by johnd0e** [comment]
Ready for review

**Comment by johnd0e** [comment]
> Could this be used for this point ?

Unfortunately docs are wrong here, and we never get error with zoom in range 1..21.

**Comment by brunob** [comment]
'k, let's merge it :)

**Comment by johnd0e** [comment]
One more note.
- Now we check max available zoom level on layer add.
- For some applications it'd more appropriate to check that continuously.
  It's not hard to implement it, but I doubt that it should be done in `L.BingLayer` itself, as everyone may implement it in own code (using provided `applyMaxNativeZoom`).

**Comment by brunob** [comment]
Nice, thx for that work @johnd0e !

**Pull request merged**